test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_fill.cpp

```diff
@@ -0,0 +1,457 @@
+                                      nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_OPERATION,
+                               "clCommandFillImageKHR should return "
+                               "CL_INVALID_OPERATION",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        dst_image = create_image_2d(context, CL_MEM_WRITE_ONLY, &formats, 512,
+                                    512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        return CL_SUCCESS;
+    }
+
+    clMemWrapper dst_image;
+    const cl_uint fill_color_1[4] = { 0x05, 0x05, 0x05, 0x05 };
+    const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+    const size_t origin[3] = { 0, 0, 0 };
+    const size_t region[3] = { 512, 512, 1 };
+    const cl_int pattern = 0xFF;
+};
+
+// CL_INVALID_VALUE if mutable_handle is not NULL.
+struct CommandBufferCommandFillMutableHandleNotNull
+    : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_mutable_command_khr mutable_handle;
+
+        cl_int error = clCommandFillBufferKHR(
+            command_buffer, nullptr, out_mem, &pattern, sizeof(cl_int), 0,
+            data_size(), 0, nullptr, nullptr, &mutable_handle);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clCommandFillBufferKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        error = clCommandFillImageKHR(command_buffer, nullptr, dst_image,
+                                      fill_color_1, origin, region, 0, nullptr,
+                                      nullptr, &mutable_handle);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clCommandFillImageKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+
+    cl_int SetUp(int elements) override
+    {
+        cl_int error = BasicCommandBufferTest::SetUp(elements);
+        test_error(error, "BasicCommandBufferTest::SetUp failed");
+
+        dst_image = create_image_2d(context, CL_MEM_WRITE_ONLY, &formats, 512,
+                                    512, 0, NULL, &error);
+        test_error(error, "create_image_2d failed");
+
+        return CL_SUCCESS;
+    }
+
+    clMemWrapper dst_image;
+    const cl_uint fill_color_1[4] = { 0x05, 0x05, 0x05, 0x05 };
+    const cl_image_format formats = { CL_RGBA, CL_UNSIGNED_INT8 };
+    const size_t origin[3] = { 0, 0, 0 };
+    const size_t region[3] = { 512, 512, 1 };
+    const cl_int pattern = 0xFF;
+};
+};
```

Comment:
It seems this is meant to terminate the anonymous namespace.
If I'm right, then there's a trailing semi-colon here.

Reply:
@silverclaw Yes, this is closing anonymous namespace. This is done in the same way as other tests in this directory

Reply:
Since an trailing semi-colon is an error, I don't think we should copy the convention. I'd prefer we didn't have it in new tests and then created a separate PR to remove it from existing tests.